src/Microsoft.AspNetCore.Authentication.Cookies/CookieExtensions.cs

```diff
@@ -15,7 +19,62 @@ public static class CookieExtensions
         public static IServiceCollection AddCookieAuthentication(this IServiceCollection services, Action<CookieAuthenticationOptions> configureOptions) =>
             services.AddCookieAuthentication(CookieAuthenticationDefaults.AuthenticationScheme, configureOptions);
 
-        public static IServiceCollection AddCookieAuthentication(this IServiceCollection services, string authenticationScheme, Action<CookieAuthenticationOptions> configureOptions) =>
-            services.AddScheme<CookieAuthenticationOptions, CookieAuthenticationHandler>(authenticationScheme, configureOptions);
+        public static IServiceCollection AddCookieAuthentication(this IServiceCollection services, string authenticationScheme, Action<CookieAuthenticationOptions> configureOptions)
+        {
+            // Makes sure that DataProtectionProvider is always set.
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IInitializeOptions<CookieAuthenticationOptions>, EnsureDataProtection>());
+            services.Initialize<CookieAuthenticationOptions>(authenticationScheme, options =>
```

Comment:
Instead of having the initialization in two places (in a delegate and in a private class), maybe it would be worth consolidating the two and moving everything to a public class? (e.g `CookieAuthenticationInitializer`)